roles/compliance/tasks/install.yml

```
@@ -0,0 +1,142 @@
+    context: "{{ cifmw_openshift_context | default(omit)}}"
+    state: present
+    wait: true
+    definition:
+      apiVersion: operators.coreos.com/v1alpha1
+      kind: Subscription
+      metadata:
+        name: compliance-operator-sub
+        namespace: "{{ cifmw_compliance_namespace }}"
+      spec:
+        name: compliance-operator
+        source: redhat-operators
+        sourceNamespace: openshift-marketplace
+        channel: "stable"
+        installPlanApproval: Automatic
+  when: not cifmw_compliance_dry_run | bool
+
+- name: Wait until the compliance-operator csv is present
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit)}}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    namespace: "{{ cifmw_compliance_namespace }}"
+    kind: ClusterServiceVersion
+    api_version: operators.coreos.com/v1alpha1
+  register: csv_list
+  delay: 10
+  retries: 20
+  until: >-
+    {{
+      csv_list.resources |
+      map(attribute='metadata.name') |
+      select('match', 'compliance-operator')
+    }}
+  when: not cifmw_compliance_dry_run | bool
+
+- name: Get full name of the compliance-operator csv
+  ansible.builtin.set_fact:
+    compliance_operator_csv_name: >-
+      {{
+        csv_list.resources |
+        map(attribute='metadata.name') |
+        select('match', 'compliance-operator') | first
+      }}
+  when: not cifmw_compliance_dry_run | bool
+
+- name: Wait for the compliance-operator csv to Succeed
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit)}}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    namespace: "{{ cifmw_compliance_namespace }}"
+    kind: ClusterServiceVersion
+    api_version: operators.coreos.com/v1alpha1
+    name: "{{ compliance_operator_csv_name }}"
+  register: csv
+  retries: 20
+  delay: 10
+  until: csv.resources[0].status.phase | default(omit) == "Succeeded"
+  when: not cifmw_compliance_dry_run | bool
+
+# Install the compliance plugin
+- name: Install podman
+  ansible.builtin.import_role:
+    name: podman
+
+- name: Log into registry
+  ansible.builtin.command:
+    cmd: "podman login -u {{ cifmw_compliance_podman_username }} -p {{ cifmw_compliance_podman_password }} {{ cifmw_compliance_podman_registry }}"
+  when: not cifmw_compliance_dry_run | bool
+
+- name: Copy out the compliance plugin from the oc-compliance image
+  ansible.builtin.command:
+    cmd: "podman run --rm -v ~/.local/bin:/mnt/out:Z {{ cifmw_compliance_plugin_image }} /bin/cp /usr/bin/oc-compliance /mnt/out/"
+  when: not cifmw_compliance_dry_run | bool
+
+# Install packages to process results
+# bzip2 comes from @rhosp-rhel-9.4-baseos
+# openscap-utils from rhosp-rhel-9.4-appstream
+- name: Install packages to process results
+  become: true
+  ansible.builtin.package:
```

we don't need to let ansible choose the package manager since we're only using EL based distro.

```suggestion
  ansible.builtin.dnf:
```

---

will do .